fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/ItemStackMixin.java

```diff
@@ -48,33 +46,17 @@
 public abstract class ItemStackMixin implements FabricItemStack {
 	@Shadow public abstract Item getItem();
 
-	@Unique
-	private LivingEntity fabric_damagingEntity;
-
-	@Unique
-	private Consumer<LivingEntity> fabric_breakCallback;
-
-	@Inject(method = "damage(ILnet/minecraft/entity/LivingEntity;Ljava/util/function/Consumer;)V", at = @At("HEAD"))
-	private void saveDamager(int amount, LivingEntity entity, Consumer<LivingEntity> breakCallback, CallbackInfo ci) {
-		this.fabric_damagingEntity = entity;
-		this.fabric_breakCallback = breakCallback;
-	}
-
-	@ModifyArg(method = "damage(ILnet/minecraft/entity/LivingEntity;Ljava/util/function/Consumer;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;damage(ILnet/minecraft/util/math/random/Random;Lnet/minecraft/server/network/ServerPlayerEntity;)Z"), index = 0)
-	private int hookDamage(int amount) {
+	@WrapOperation(method = "damage(ILnet/minecraft/entity/LivingEntity;Lnet/minecraft/entity/EquipmentSlot;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;damage(ILnet/minecraft/util/math/random/Random;Lnet/minecraft/server/network/ServerPlayerEntity;Ljava/lang/Runnable;)V"))
+	private void hookDamage(ItemStack instance, int amount, Random random, ServerPlayerEntity serverPlayerEntity, Runnable runnable, Operation<Void> original) {
 		CustomDamageHandler handler = ((ItemExtensions) getItem()).fabric_getCustomDamageHandler();
 
 		if (handler != null) {
-			return handler.damage((ItemStack) (Object) this, amount, fabric_damagingEntity, fabric_breakCallback);
+			// TODO 1.20.5, what to do about break callback
+			assert false;
+			amount = handler.damage((ItemStack) (Object) this, amount, serverPlayerEntity, null);
```

Comment:
Can't you just pass the `Runnable` like this for now, as the previous behaviour just seemed to use the vanilla break callback?
```suggestion
			amount = handler.damage((ItemStack) (Object) this, amount, serverPlayerEntity, e -> runnable.run());
```

As a side note: deprecating the `CustomDamageHandler::damage` method and using a new one would be a bit odd, as it would still require people to upgrade to the new one, so I don't think that would be that viable? For example, this
```java
CustomDamageHandler handler = (stack, amount, entity, breakCallback) -> {
	breakCallback.accept(entity);
	return amount;
};
```
would still have to be changed to this
```java
CustomDamageHandler handler = (stack, amount, entity, breakCallback) -> {
	breakCallback.run();
	return amount;
};
```
Then again, `FabricItemSettings` is getting deprecated too in #3534, so maybe it could be a different case here?

Reply:
We don't mind breaking changes during port if it makes sense.

Reply:
This was resolved.